packages/prism-react-renderer/src/components/useTokenize.ts

```diff
@@ -21,9 +20,15 @@ export const useTokenize = ({ prism, code, grammar, language }: Options) => {
       tokens: [],
     }
 
-    prismRef.current.hooks.run("before-tokenize", prismConfig)
-    prismConfig.tokens = prismRef.current.tokenize(code, grammar)
-    prismRef.current.hooks.run("after-tokenize", prismConfig)
+    prism.hooks.run("before-tokenize", prismConfig)
+    prismConfig.tokens = prism.tokenize(code, grammar)
+    prism.hooks.run("after-tokenize", prismConfig)
     return normalizeTokens(prismConfig.tokens)
-  }, [code, grammar, language])
+  }, [
+    code,
+    grammar,
+    language,
+    // prism is a stable import
+    prism,
```

Comment:
since prism is being imported, it's stable and won't cause unnecessary recalculations. I tested this a bit by changing some state while `prism` was in the dependency array and the memoized function was not invoked again